skyfield/keplerlib.py

```diff
@@ -374,9 +375,21 @@ def ele_to_vec(p, e, i, Om, w, v, mu):
 
 dpmax = sys.float_info.max
 
+
 def bracket(num, end1, end2):
-    num[num<end1] = end1
-    num[num>end2] = end2
+    low = num<end1
+    high = num>end2
+
+    num[low] = repeat(end1, sum(low, axis=1))
+    num[high] = repeat(end2, sum(high, axis=1))
+    return num
+
+
+def bracket_1d(num, end1, end2):
+    too_low = num < end1
```

Comment:
Try taking a look at `np.where()` — I think here it could eliminate the need to build 3 intermediate arrays.

Reply:
This comment inspired me to discover the existence of `np.clip`, which already does what the bracket functions do. Hopefully it can replace both of them!

Reply:
In newer versions of numpy this line:
```
x[past] = bracket_1d(lower[past], repeat(-bound, orb_ind), repeat(bound, orb_ind))
```
can be changed to
```
clip(lower, -bound, bound, where=past, out=x)
```
which is _so_ nice, and also means we don't need `bracket_1d` at all. Unfortunately in the version of numpy pinned in `requirements.txt` `clip` was not yet turned into a ufunc so `where` and `out` are not valid keywords. I'm assuming that means it's a no go?

Reply:
Can you `clip()` into an intermediate array, and then do the where/out using normal assignment? (`x[past] = clipped`)

I generally keep NumPy as old as possible, to keep to a minimum the number of issues opened by folks when I accidentally use NumPy features that are too recent. I have no actual data on what versions folks are using, so if, for example, the last user of NumPy 1.11 were to shut down their old project, I would never get notified that I could advance to 1.12, so the system is not a perfect one. But I as I prefer to avoid leaving anyone behind, let’s see if we can continue forward with 1.11 for now.

Reply:
I was able to replace `bracket_1d` directly with `clip` like you suggested. 

I was also able to replace `bracket` with 2 calls to `np.copyto`, which was added in numpy 1.7 according to the documentation.